dags/aps/aps.py

```diff
@@ -0,0 +1,44 @@
+import os
+import warnings
+
+import airflow
+from airflow.decorators import dag, task
+from aps.APSApiClient import APSApiClient
+from aps.APSParams import APSParams
+from aps.fetch_from_api import save_file_in_s3, set_APS_harvesting_interval, split_json
+from aps.fetch_from_api import trigger_file_processing as trigger_file_processing_
+from aps.repository import APSRepository
+from common.repository import IRepository
+
+
+@dag(start_date=airflow.utils.dates.days_ago(0))
+def aps_fetch_api():
+    @task()
+    def set_fetching_intervals(repo: IRepository = APSRepository()):
+        return set_APS_harvesting_interval(repo=repo)
+
+    @task()
+    def save_json_in_s3(dates: dict, repo: IRepository = APSRepository()):
+        parameters = APSParams(
+            from_date=dates["aps_fetching_from_date"],
+            until_date=dates["aps_fetching_until_date"],
+        ).get_params()
+        rest_api = APSApiClient(
+            base_url=os.getenv("APS_BASE_URL", "http://harvest.aps.org")
+        )
+        return save_file_in_s3(rest_api=rest_api, repo=repo, parameters=parameters)
```

Comment:
the logic to fetch the JSON from the rest api should belong here. `save_file_in_s3` shouldn't have the logic to interact with the rest API, the sole responsibility of this function is to save the file ([more details](https://en.wikipedia.org/wiki/Single-responsibility_principle)).